### ISSUE TITLE
How to read PlayerInputChange?

### ISSUE THREAD

**Issue by goteppo** [Trey2k/RLBotGo#10]
Hi. I was trying to get a bot to to double jump (or flip) as quickly as possible, and intended to use PlayerInputChange to track what the game thinks our jump-button state is, but couldn't find any way to read the PlayerInputChange events.

Is this something that's not fully implemented yet, or did I miss how PlayerInputChange is intended to be used?

Without this information, it seems that I need to add some arbitrary delay between the release of the jump button and the second jump, but it seems slightly suboptimal, and reducing the delay too much seems to cause the second jump to not always register.

Has anyone else dealt with the same issue? Thanks!

**Comment by Trey2k** [comment]
Just checked it seems that the GameMessagePacket was never added some how
As far as I can tell this is what is missing. Hopefully I will get some time here shortly to resolve this Thanks for bringing this up!
```
table PlayerInputChange
{
	playerIndex: int;
	controllerState: ControllerState;
	dodgeForward:float;
	dodgeRight:float;
}

union GameMessage
{
    PlayerStatEvent,
    PlayerSpectate,
	PlayerInputChange
}

table GameMessageWrapper
{
	Message:GameMessage;
}

table MessagePacket
{
	messages:[GameMessageWrapper];
        gameSeconds: float;
        frameNum: int;
}
```

**Comment by Trey2k** [comment]
Fixed in 3001c90b7fe25287c8d9ad98555f71057adce461

Game messages can be accessed via the gameState now. This is what that looks like:
```go
func getInput(gameState *RLBot.GameState, rlBot *RLBot.RLBot) *RLBot.ControllerState {

	if gameState.GameMessageOK {
		for _, msg := range gameState.GameMessage.Messages {
			switch msg.GameMessageType {
			case RLBot.GamesMessageType_PlayerInputChange:
				fmt.Println("PlayerInputChange")
				fmt.Printf("%d\n", msg.PlayerInputChange.PlayerIndex)
			case RLBot.GamesMessageType_PlayerSpectate:
				fmt.Println("PlayerSpectate")
				fmt.Printf("%d\n", msg.PlayerSpectate.PlayerIndex)
			case RLBot.GamesMessageType_PlayerStatEvent:
				fmt.Println("PlayerStatEvent")
				fmt.Printf("%d: %s\n", msg.PlayerStatEvent.PlayerIndex, msg.PlayerStatEvent.StatType)
			}
		}
	}

	PlayerInput := &RLBot.ControllerState{}
	return PlayerInput

}
```

**Comment by goteppo** [comment]
Thank you for the quick fix @Trey2k - much appreciated!
